strongswan: support /var/swanctl/conf.d/*.conf

hgl opened the pull request:
Maintainer: @pprindeville @Thermi 
Compile tested: x86 master
Run tested: x86 master

Description:

Sometimes, users might want to generate dynamic conf files. One example would be creating a IPv6 pool using WAN PD that might change. Currently, only /etc/swanctl/conf.d/*.conf is supported for user conf files, but it's not a good fit for dynamic ones, since upon rebooting the device, the conf can become stale.

With this change, users, for example, can use hotplug to create the conf in /var/swanctl/conf.d and have swanctl correctly load it.

They could of course add the include directive in
/etc/swanctl/swanctl.conf, but then they have to also remember creating the /var/swanctl/conf.d directory before every generation of the dynamic conf, which can be cumbersome. Given OpenWrt's popularity in home network, direct support for dynamic user conf files should be desirable.

pprindeville commented on net/strongswan/files/swanctl.init:
```diff
@@ -654,6 +654,8 @@ config_ipsec() {
 }
 
 do_postamble() {
+	swanctl_xappend0 "include /var/swanctl/conf.d/*.conf"
```
Why before the files in `/etc` and not after?

hgl replied:
swanctl_xappend0 manipulates `/var/swanctl/swanctl.conf`, lines down below modifies `/var/ipsec/strongswan.conf`, I'm not sure I understand what files `/etc` are you referring to?

hgl committed:
strongswan: support /var/swanctl/conf.d/*.conf

Sometimes, users might want to generate dynamic conf files. One example
would be creating a IPv6 pool using WAN PD that might change. Currently,
only /etc/swanctl/conf.d/*.conf is supported for user conf files, but
it's not a good fit for dynamic ones, since upon rebooting the device,
the conf can become stale.

With this change, users, for example, can use hotplug to create the conf
in /var/swanctl/conf.d and have swanctl correctly load it.

They could of course add the include directive in
/etc/swanctl/swanctl.conf, but then they have to also remember creating
the /var/swanctl/conf.d directory before every generation of the dynamic
conf, which can be cumbersome. Given OpenWrt's popularity in home
network, direct support for dynamic user conf files should be desirable.

Signed-off-by: Glen Huang <me@glenhuang.com>

hgl commented:
@pprindeville I've moved the include directive to Makefile, hope that's a better place for this logic.

pprindeville reviewed (changes requested):
Needs `PKG_RELEASE` bump.

hgl commented:
Since I bumped in the other PR, do you think this PR can be merged with the other PR? Bumping at both can create conflicts.

BKPepe commented:
Conflicts can not be resolved? They can, so bump the version and lets merge this one.:-)

hgl commented:
Once https://github.com/openwrt/packages/pull/22716 is merged, this one should be directly mergeable.

1715173329 commented:
please rebase this PR, thanks!

zcracker commented:
please rebase this PR

Neustradamus commented:
@hgl: Any progress on your PR?
Soon one year...

pprindeville reviewed (changes requested):
And a `PKG_RELEASE` bump and I'll merge it.
